### PR TITLE
Pandas stub fixes

### PR DESCRIPTION
**Pull request by gramster**
Add a few missing files.
Update .gitignore.
Make some fixes based on the tests from the pandas-stubs package (more to come).

### CODE REVIEW

File: doc/pandas/core/frame.pyi.ds [hunk 1]
```diff
@@ -1332,7 +1333,6 @@ observed : bool, default False
 
 Returns
 -------
-DataFrameGroupBy
```
Comment (bschnurr) [inline doc/pandas/core/frame.pyi.ds:1335]: `DataFrameGroupBy` removed on purpose?

**Comment by yanyongyu** [comment]
`pandas.io.parser.TextFileReader` now support `context manager` with `__enter__` and `__exit__` function but not include in stub file

https://github.com/microsoft/python-type-stubs/blob/e7def7eb38b32fdc1d5b9ae9ad4df25b8696b97e/pandas/io/parsers.pyi#L553-L564

**Review by bschnurr — APPROVED** [review]

**Comment by gramster** [comment]
I don't think it's this change; there was a reversion to `IO` that happened in the copy in pylance that has since been fixed but must have made it out in the release. This change already includes that reversion.

**Comment by Dr-Irv** [comment]
@gramster even if I do the change above, then the following test fails:
```python
from io import StringIO
import pandas as pd
from pandas.io.parsers import TextFileReader

dio = StringIO("a,b\n 1,2\n 3,4\n")

df: pd.DataFrame = pd.read_csv(dio)

print(df)

dio.seek(0)
tr: pd.DataFrame = pd.read_csv(dio)

dio.seek(0)
tr2: TextFileReader = pd.read_csv(dio, iterator=True)

dio.seek(0)
tr3: pd.DataFrame = pd.read_csv(dio, iterator=False)

dio.seek(0)
tr5: TextFileReader = pd.read_csv(dio, chunksize=10)

dio.seek(0)
tr6: pd.DataFrame = pd.read_csv(dio, chunksize=None)

dio.seek(0)
tr7: pd.DataFrame = pd.read_csv(dio, header="infer")

dio.seek(0)
tr8: TextFileReader = pd.read_csv(dio, header="infer", iterator=True)


ftr1: TextFileReader = pd.read_csv("hey.csv")
ftr2: pd.DataFrame = pd.read_csv("hey.csv", iterator=True)
ftr3: TextFileReader = pd.read_csv("hey.csv", iterator=False)
ftr5: pd.DataFrame = pd.read_csv("hey.csv", chunksize=10)
ftr6: TextFileReader = pd.read_csv("he.csv", chunksize=None)
ftr7: TextFileReader = pd.read_csv("hey.csv", header="infer")
ftr8: pd.DataFrame = pd.read_csv("hey.csv", header="infer", iterator=True)
```
All the assignments to variables starting with `tr` should be legal, while the assignments to the ones using `ftr` should fail.

The deletion of the `*` argument at the beginning is what I believe to be the cause of those failures.

**Pull request merged**